src/components/generic/form-autocomplete.tsx

```diff
@@ -0,0 +1,323 @@
+import { Metaform, MetaformField, MetaformFieldAutocompleteService, MetaformFieldSourceType } from "../../generated/client";
+import { Attribute, Qfield } from "../../generated/codeserver-client";
+import { FieldValue } from "metaform-react";
+import Config from "../../config";
+import strings from "../../localization/strings";
+import styles from "../../styles/form";
+import { autocompleteErrorMessages, AutocompleteItem } from "../../types";
+
+/**
+ * Interface representing component properties
+ */
+interface Props extends WithStyles<typeof styles> {
+  minSearchLength: number;
+  searchInterval: number;
+  metaform: Metaform;
+  field: MetaformField;
+  disabled: boolean;
+  value: FieldValue;
+  setFieldValue: (fieldName: string, fieldValue: FieldValue) => void;
+}
+
+/**
+ * Interface representing component state
+ */
+interface State {
+  loading: boolean;
+  items: AutocompleteItem[];
+  inputValue: string;
+  defaultValue?: AutocompleteItem;
+  errorMessage: string;
+}
+
+/**
+ * React component displaying form autocomplete fields
+ */
+export default class FormAutocomplete extends React.Component<Props, State> {
+
+  /**
+   * Constructor
+   * 
+   * @param props component properties
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      loading: false,
+      items: [],
+      inputValue: "",
+      errorMessage: ""
+    };
+  }
+
+  /**
+   * Component did mount life cycle event
+   */
+  public componentDidMount = async () => {
+    const { value, setFieldValue } = this.props;
+
+    this.setState({
+      loading: true
+    });
+
+    try {
+      const loadedItems = await this.loadItems();
+      const defaultAutoCompleteItem = loadedItems.find(item => item.id === value as string);
+
+      this.setState({
+        items: loadedItems,
+        loading: false,
+        defaultValue: defaultAutoCompleteItem ? { ...defaultAutoCompleteItem } : defaultAutoCompleteItem
+      });
+
+      this.getSourceFields().forEach(sourceField => {
+        if (sourceField.name) {
+          const itemProperty = sourceField.source?.options?.autocompleteItemProperty;
+          const itemValue = itemProperty && defaultAutoCompleteItem ? defaultAutoCompleteItem[itemProperty] : null;
+          setFieldValue(sourceField.name, itemValue);
+        }
+      });
+    } catch (e) {
+      this.setState({        
+        loading: false,
+        errorMessage: e.message
+      });
+    }
+    
+  }
+
+  /** 
+   * Component render method
+   */
+  public render() {
+    const {
+      field,
+      disabled,
+      value,
+      classes
+    } = this.props;
+
+    const {
+      items,
+      inputValue,
+      defaultValue,
+      loading,
+      errorMessage
+    } = this.state;
+
+    const selectedAutocompleteItem = items.find(item => item.id === value as string);
+
+    if (errorMessage) {
+      return this.renderErrorMessage();
+    }
+
+    if (loading || !items && !errorMessage) {
+      return this.renderLoader();
```

Comment:
The existence of items or error message is defining if the loader is showed or not, so I think we might no even need the loading state in this component.

Reply:
The autocomplete loads all the autocomplete during mount and set the default item after loading the options, I think it would be logical to prevent user interaction before the component is fully setup.